common/db/workspace.go

```diff
@@ -0,0 +1,200 @@
+package db
+
+import (
+	"GADS/common/models"
+	"context"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func AddWorkspace(workspace *models.Workspace) error {
+	workspace.GenerateUUID()
+	collection := mongoClient.Database("gads").Collection("workspaces")
+	_, err := collection.InsertOne(mongoClientCtx, workspace)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func UpdateWorkspace(workspace *models.Workspace) error {
+	collection := mongoClient.Database("gads").Collection("workspaces")
+	filter := bson.M{"_id": workspace.ID}
+	update := bson.M{"$set": workspace}
+	_, err := collection.UpdateOne(mongoClientCtx, filter, update)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func DeleteWorkspace(id string) error {
+	collection := mongoClient.Database("gads").Collection("workspaces")
+	filter := bson.M{"_id": id}
+	_, err := collection.DeleteOne(mongoClientCtx, filter)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetWorkspaces() []models.Workspace {
+	var workspaces []models.Workspace
+	collection := mongoClient.Database("gads").Collection("workspaces")
+
+	cursor, err := collection.Find(mongoClientCtx, bson.M{})
+	if err != nil {
+		return workspaces
+	}
+	defer cursor.Close(mongoClientCtx)
+
+	for cursor.Next(mongoClientCtx) {
+		var workspace models.Workspace
+		if err := cursor.Decode(&workspace); err != nil {
+			continue
+		}
+		workspaces = append(workspaces, workspace)
+	}
+
+	return workspaces
+}
+
+func WorkspaceHasDevices(id string) bool {
+	collection := mongoClient.Database("gads").Collection("new_devices")
+	filter := bson.M{"workspace_id": id}
+	count, err := collection.CountDocuments(mongoClientCtx, filter)
+	if err != nil {
+		return false
+	}
+	return count > 0
+}
+
+func WorkspaceHasUsers(id string) bool {
+	collection := mongoClient.Database("gads").Collection("users")
+	filter := bson.M{"workspace_ids": id}
```

Comment:
Will this filter work properly if we have multiple workspaces since its an array? Not really sure, just asking :D

Reply:
Yes, when we query an array field with a single value, MongoDB will match any document where the array contains that value ([reference](https://www.mongodb.com/docs/manual/tutorial/query-arrays/#query-an-array-for-an-element))